### PR TITLE
Update chakra theme and fix design-related bugs #36

### LINKED ISSUE #36: Update chakra theme
### Add
- [x] rem size and font size
- [x] css variables cheat sheet
- [x] explicitly show that themed components like `CircularProgress` is used differently

### Style
- [x] Accordion
  - [x] rounded
  - [x] hover
- [x] Tooltip
- [x] Make tags readmore button wider

### Fix
- [x] line height
- [x] change focus to focus-visible
- [x] tabs :active dim background
- [x] tabs bottom border color and width

### PR DESCRIPTION
**Pull request by moetazaneta**
closes #36.

We use 4px-based design system, so if we ever need to hardcode pixels (ReleasePage.tsx, ln 102, col 32), it must be divisible by 4.
Overflow of characters cards in overview tabs, will be addressed in #66.
Not 4px-based value in CharacterCard.tsx (line 31) is intended.

For css variables intellisense I suggest [CSS Variable Autocomplete](https://marketplace.visualstudio.com/items?itemName=vunguyentuan.vscode-css-variables), we will need to create file with vars and include it in extension's config, want feedback.

### CODE REVIEW

**Comment by prekel** [comment]
pls name pr

File: src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.module.css [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
 
 .info {
   display: flex;
   flex-direction: column;
-  gap: 24px;
+  gap: var(--chakra-sizes-6);
 }
 
 .image {
-  height: 300px;
```
Comment (Moltenship) [inline src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.module.css:29]: Is this intentional? We are breaking image style with this `height: auto`

Comment (moetazaneta) [reply src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.module.css:29]: I add this so the image have proper aspect ratio. Also I made boundary if the image is too tall.
I think we would need to set height for vn header block, when implementing "read more" functionality for description.

File: src/theme/components/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+interface StyleOptions {
+  theme: Theme;
+  colorMode: 'light' | 'dark';
+  colorScheme: string;
+}
+
+export interface StyleConfig {
+  baseStyle?: StyleInterpolation;
+  sizes?: Record<string, StyleInterpolation>;
+  variants?: Record<string, StyleInterpolation>;
```
Comment (NightOfNights) [inline src/theme/components/types.ts]: I think you should create types for `size` and `variant`  and narrow `string` there to these types

Comment (NightOfNights) [reply src/theme/components/types.ts]: To avoid missing key error we can wrap `Record<...>` in `Partial`

**Review by NightOfNights — COMMENTED** [review]
Do not forget add task number to your pr name 🥲

**Comment by NightOfNights** [comment]
It seems inactive tabs have become unfocusable. Is it intended?

File: src/features/VisualNovel/components/VisualNovelTabs/VisualNovelTabs.tsx [hunk 3]
```diff
@@ -22,9 +22,7 @@ export const VisualNovelTabs: FC<VisualNovelTabsProps> = memo(({ id }) => {
   const location = useLocation();
 
   const tabs = visualNovelTabInfo.map(tabInfo => (
-    <Link key={tabInfo.name} to={`/vn/${id}/${tabInfo.path}`}>
-      <Tab>{tabInfo.name}</Tab>
-    </Link>
+    <Tab as={Link} key={tabInfo.name} to={`/vn/${id}/${tabInfo.path}`}>{tabInfo.name}</Tab>
```
Comment (NightOfNights) [inline src/features/VisualNovel/components/VisualNovelTabs/VisualNovelTabs.tsx:25]: It seems inactive tabs have become unfocusable after this change

File: src/features/VisualNovel/pages/ReleasesPage/ReleasesPage.tsx [hunk 4]
```diff
@@ -115,7 +116,7 @@ export const ReleasesPage: VFC = () => {
                   return (
                     <Tooltip
                       key={platform + String(release.id)}
-                      hasArrow
+
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/ReleasesPage/ReleasesPage.tsx]: remove gap between lines

File: src/features/VisualNovel/pages/ReleasesPage/ReleasesPage.tsx [hunk 5]
```diff
@@ -129,7 +130,7 @@ export const ReleasesPage: VFC = () => {
                 {ReleaseService.getReleaseIcons(release).map(releaseIcon => (
                   <Tooltip
                     key={releaseIcon.icon + releaseIcon.label}
-                    hasArrow
+
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/ReleasesPage/ReleasesPage.tsx]: same

**Comment by NightOfNights** [comment]
Pls update VisualNovelTabs component. With this changes it would apply some styles ONLY to active tab

<details><summary>Updated component</summary>

```js
import React, { VFC, memo, useState } from 'react';
import { Tabs, TabList, Tab } from '@chakra-ui/react';
import { Link, useLocation } from 'react-router-dom';
import { visualNovelTabInfo } from '../../utils/constants';
import cls from './VisualNovelTabs.module.css';

interface Props {

  /**
   * Id of visual novel.
   */
  id: string;
}

const FIRST_INDEX = 0;

/**
 * Calculates initial index for tabs based on route.
 * @param pathname URL pathname.
 */
const getInitialTabIndex = (pathname: string): number => {
  const splitPath = pathname.split('/');
  const activeRoute = splitPath[splitPath.length - 1];
  const tabIndex = visualNovelTabInfo.findIndex(tabInfo => tabInfo.path === activeRoute);
  return tabIndex >= FIRST_INDEX ? tabIndex : FIRST_INDEX;
};

/**
 * Component for navigation on visual novel page.
 * TODO (Panov A.): Add badges to tabs.
 */
export const VisualNovelTabs: VFC<Props> = memo(({ id }) => {
  const location = useLocation();
  const [tabIndex, setTabIndex] = useState(getInitialTabIndex(location.pathname));

  /**
   * Handles click on tab.
   * @param index Tab index.
   */
  const onTabClick = (index: number): () => void => () => setTabIndex(index);

  /**
   * Handles key press on tab.
   * @param index Tab index.
   */
  const onTabPress = (index: number): (event: React.KeyboardEvent) => void => (event: React.KeyboardEvent) => {
    if (event.key === 'Enter') {
      setTabIndex(index);
    }
  };

  const tabs = visualNovelTabInfo.map((tabInfo, index) => (
    <Tab
      as={Link}
      key={tabInfo.name}
      to={`/vn/${id}/${tabInfo.path}`}
      onClick={onTabClick(index)}
      onKeyPress={onTabPress(index)}
    >
      {tabInfo.name}
    </Tab>
  ));

  return (
    <nav>
      <Tabs index={tabIndex} colorScheme="orange">
        <TabList className={cls.tabList}>
          {tabs}
        </TabList>
      </Tabs>
    </nav>
  );
});

```
</details>

File: src/theme/components/types.ts (hunk 2 above)
Comment (moetazaneta) [reply src/theme/components/types.ts]: I copy it from chakra documentation,
I don't think `variant` should be narrowed, for example if I added `no-underline` for links, I would need to update types.
For the `size` I will narrow it down.

**Review by NightOfNights — APPROVED** [review]

**Review by Moltenship — APPROVED** [review]

**Review by NightOfNights — APPROVED** [review]

**Review by prekel — APPROVED** [review]

**Pull request merged**